routers/api/v1/repo/mirror.go

```diff
@@ -63,3 +74,302 @@ func MirrorSync(ctx *context.APIContext) {
 
 	ctx.Status(http.StatusOK)
 }
+
+// PushMirrorSync adds all push mirrored repositories to the sync queue
+func PushMirrorSync(ctx *context.APIContext) {
+	// swagger:operation POST /repos/{owner}/{repo}/push_mirrors-sync repository repoPushMirrorSync
+	// ---
+	// summary: Sync all push mirrored repository
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo to sync
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo to sync
+	//   type: string
+	//   required: true
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/empty"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+
+	repo := ctx.Repo.Repository
+
+	if !setting.Mirror.Enabled {
+		ctx.Error(http.StatusBadRequest, "PushMirrorSync", "Mirror feature is disabled")
+		return
+	}
+	// Get All push mirrors of a specific repo
+	pushMirrors, err := repo_model.GetPushMirrorsByRepoID(repo.ID, db.ListOptions{})
+	if err != nil {
+		ctx.Error(http.StatusBadRequest, "PushMirrorSync", err)
+		return
+	}
+	for _, mirror := range pushMirrors {
+		mirror_service.SyncPushMirror(ctx, mirror.ID)
```

Comment:
this functions returns `false` on error. Don't thin we should return `StatusOK` when some mirror failed.

Reply:
Should we stop the sync of the other mirrors and return an error when a mirror fails? 

```suggestion
		ok := mirror_service.SyncPushMirror(ctx, mirror.ID)
		if !ok {
			ctx.Error(http.StatusBadRequest, "PushMirrorSync", "error occured when syncing push mirror "+mirror.RemoteName)
		}
```

or try to sync the other push mirrors then throw the error at the end

``` go
		syncError := false
	        for _, mirror := range pushMirrors {
		        ok := mirror_service.SyncPushMirror(ctx, mirror.ID)
		        if !ok {
			        syncError = true
		        }
	        }
	        if syncError {
		        ctx.Error(http.StatusBadRequest, "PushMirrorSync", "error occured when syncing push mirrors")
	        }
```
```

Reply:
It's seems to be consistent with other API's to stop the syncing of the mirrors and return a internal server error.